app/uk/gov/hmrc/mtdsatestsupportapi/models/hateoas/CheckpointHateoasLinks.scala

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.mtdsatestsupportapi.models.hateoas
+
+import api.hateoas.Link
+import config.AppConfig
+import api.hateoas.Method._
+import api.models.domain.Nino
+import uk.gov.hmrc.mtdsatestsupportapi.models.common.CheckpointId
+
+trait CheckpointHateoasLinks {
+
+  val SELF = "self"
+
+  private def baseUrl(appConfig: AppConfig) =
+    s"/${appConfig.apiGatewayContext}/vendor-state/checkpoints"
+
+  def createCheckpoint(appConfig: AppConfig, nino: Nino): Link =
+    Link(href = s"${baseUrl(appConfig)}/?nino=${nino.value}", method = POST, rel = "create-checkpoint")
+
+  def restoreCheckpoint(appConfig: AppConfig, checkpointId: CheckpointId): Link =
```

Comment:
I think the restore url should be `href = s"${baseUrl(appConfig)}/restore/${checkpointId.value}"`, or does the order not matter here?

Reply:
I think the spec is wrong -- from a REST viewpoint the restore is below the checkpoint's own resource. (I will fix spec). The code agrees with the HATEOAS spec.